### PR TITLE
remove deprecations for 1.0 release

### PR DESCRIPTION
**Pull request by bkamins**

### CODE REVIEW

**Comment by bkamins** [comment]
re-trigger CI, as on nightly strange `OutOfMemory` error happens in multi-threaed code that I cannot reproduce

File: src/other/index.jl [hunk 1]
```diff
@@ -218,7 +218,7 @@ end
     setdiff(1:length(x), getindex(x, notidx.skip))
 @inline Base.getindex(x::AbstractIndex, idx::Between) = x[idx.first]:x[idx.last]
 @inline Base.getindex(x::AbstractIndex, idx::All) =
-    isempty(idx.cols) ? (1:length(x)) : union(getindex.(Ref(x), idx.cols)...)
+    isempty(idx.cols) ? (1:length(x)) : throw(ArgumentError("All(args...) is not supported"))
```
Comment (nalimilan) [inline src/other/index.jl:221]:
```suggestion
    isempty(idx.cols) ? (1:length(x)) : throw(ArgumentError("All(args...) is not supported: use Cols(args...) instead"))
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
This out-of-memory error https://github.com/JuliaData/DataFrames.jl/pull/2679/checks?check_run_id=2209199932 randomly happens on nightly.

@nalimilan - to whom do you think we should report it? It is related to multi-threading.

**Comment by nalimilan** [comment]
Maybe it's just that the VM has a limited amount of RAM? We didn't had tests with several million rows until recently. We could use 3 or even 2 columns instead of 4 there.

**Comment by bkamins** [comment]
OK. I will merge this and open a PR reducing the memory use.

**Pull request merged**